docs/customruns.md

```diff
@@ -0,0 +1,413 @@
+      (optionally) name of the custom task type to execute.
+    - [`customSpec`](#2-specifying-the-target-custom-task-by-embedding-its-spec) - Embed the custom task resource spec
+      directly in a `CustomRun`.
+- Optional:
+  - [`timeout`](#specifying-timeout) - specifies the maximum duration of a single execution
+    of a `CustomRun`.
+  - [`retries`](#specifying-retries) - specifies the number of retures upon `CustomRun`failure.
+  - [`params`](#specifying-parameters) - Specifies the desired execution
+    parameters for the custom task.
+  - [`serviceAccountName`](#specifying-a-serviceaccount) - Specifies a `ServiceAccount`
+    object that provides custom credentials for executing the `CustomRun`.
+  - [`workspaces`](#specifying-workspaces) - Specifies the physical volumes to use for the
+    [`Workspaces`](workspaces.md) required by a custom task.
+
+[kubernetes-overview]:
+  https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields
+
+### Specifying the target Custom Task
+
+A custom task resource's `CustomSpec` may be directly embedded in the `CustomRun` or it may
+be referred to by a `CustomRef`. But, not both at the same time.
+
+1. [Specifying the target Custom Task with customRef](#1-specifying-the-target-custom-task-with-customref)
+   Referring a custom task (i.e. `CustomRef` ) promotes reuse of custom task definitions.
+
+2. [Specifying the target Custom Task by embedding its spec](#2-specifying-the-target-custom-task-by-embedding-its-spec)
+   Embedding a custom task (i.e. `CustomSpec` ) helps in avoiding name collisions with other users within the same namespace.
+   Additionally, in a pipeline with multiple embedded custom tasks, the details of entire pipeline can be fetched in a
+   single API request.
+
+#### Specifying the target Custom Task with customRef
+
+To specify the custom task type you want to execute in your `CustomRun`, use the
+`customRef` field as shown below:
+
+```yaml
+spec:
+  customRef:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+```
+
+This initiates the execution of a `CustomRun` of a custom task of type `Example`, in
+the `example.dev` API group,  with the version `v1alpha1`.
+
+You can also specify the `name` and optional `namespace` (default is `default`)
+of a custom task resource object previously defined in the cluster.
+
+```yaml
+spec:
+  customRef:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    name: my-example
+```
+
+If the `customRef` specifies a name, the custom task controller should look up the
+`Example` resource with that name, and use that object to configure the
+execution.
+
+If the `customRef` does not specify a name, the custom task controller might support
+some default behavior for executing unnamed tasks.
+
+In either case, if the named resource cannot be found, or if unnamed tasks are
+not supported, the custom task controller should update the `CustomRun`'s status to
+indicate the error.
```

Comment:
Could this be separated into a "Developer Guide for supporting CustomRef section"?

Reply:
Agreed, this would be part of the doc separation, will be tracked by https://github.com/tektoncd/pipeline/issues/5775, I'd prefer to merge this PR in to get some momentum XD, but I can also be convinced that we should do it in this PR~

Reply:
I'd prefer to do it in this PR, because the docs are being introduced in this PR, and because I don't think there's anything directly blocked by these docs right now, but could be convinced otherwise.

Reply:
If I'm not mistaken, per [the Pipelines WG](https://docs.google.com/document/d/1AfJfdyd1JN2P4haBdYOxEn6SMENMgvQF9ps7iF2QSG0/edit?resourcekey=0-lm66bE68HA_WVLA6AaCh6w#bookmark=id.kzvlbp8pb72) last Tuesday, we need to have documentation ready before releasing `v1beta1.CustomTask` in `v0.42` right?

Reply:
I can also try my best to separate the guide before v0.42 :) But I'm not sure if that can be merged in before scheduled release 🤔

Reply:
Thanks Lee~ @lbernick I'd prefer we make incremental updates :P